src/Microsoft.Restier.Core/Conventions/ConventionBasedQueryExpressionProcessor.cs

```diff
@@ -149,7 +149,42 @@ private Expression AppendOnFilterExpression(QueryExpressionContext context, stri
                 var collectionType = context.VisitedNode.Type.FindGenericType(typeof(ICollection<>));
                 if (collectionType == null)
                 {
-                    return null;
+                    var nullableType = context.VisitedNode.Type;
```

Comment:
So I'm trying to implement this change in my branch, and I'm never hitting this line of code if I have "$count=true" in the URL. Have you tried this with the very latest OData binaries (7.3)?

Thanks!